code/modules/vehicle/tank.dm

```diff
@@ -0,0 +1,81 @@
+/obj/vehicle/tank
+	name = "tank"
+	icon = 'icons/vehicles/tanks.dmi'
+	icon_state = "tank_green"
+	vehicle_move_delay = 3
+	var/allow_fire = 0
+	var/recoil_delay = 10
+	var/resume_move = 10
+	var/datum/action/innate/fire_cannon/fire_cannon_action = new
+
+/obj/vehicle/tank/handle_vehicle_layer()
+	layer = MOB_LAYER+0.1
+
+
+/obj/vehicle/tank/proc/GrantActions(mob/living/carbon/buckled_mob)
+	if(allow_fire == 1)
+		fire_cannon_action.target = src
+		fire_cannon_action.Grant(buckled_mob)
+
+
+/datum/action/innate/fire_cannon
+	name = "Fire cannon"
+	button_icon_state = "fireball"
+	var/next_firetime = 50
+
+/datum/action/innate/fire_cannon/Activate()
+	var/obj/vehicle/tank/T = target
+	var/bullet_type = /obj/item/projectile/bullet/gyro
+	var/turf/curloc = get_turf(target)
+	var/fire_delay = 50
+
+	if(next_firetime > world.time)
+		to_chat(owner, "<span class='warning'>Your weapons are recharging.</span>")
+
+	else
+		var/obj/item/projectile/A = new bullet_type(curloc)
+		if(target.dir == NORTH)
+			A.Angle = 360
+
+		if(target.dir == EAST)
+			A.Angle = 90
+
+		if(target.dir == SOUTH)
+			A.Angle = 180
+
+		if(target.dir == WEST)
+			A.Angle = 270
+		A.fire()
```

Comment:
I thinkkkkk the way the experimentor handles the fireball may be more applicable here.